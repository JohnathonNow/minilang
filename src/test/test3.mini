var L := [1, 2, "hello", 4, 5]

do
	for X in L do
		print('{X} + 1 = {X + 1}\n')
		if X = 4 then
			exit 1 + "a"
		end
	on Error do
		print('Correct! An error has occurred 1: {Error:type}\n')
	end
on Error do
	print('Correct! An error has occurred 2: {Error:type}\n')
end

print('Before X loop\n')
for X in 1 .. 5 do
	print('Before Y loop\n')
	for Y in 11 .. 15 do
		print('X = {X}, Y = {Y}\n')
		if X = 3 and Y = 13 then
			exit next
		end
	end
	print('After Y loop\n')
end

for Z in 1 .. 10 do
	print('Z = {Z}\n')
	if Z = 5 then
		print('5!\n')
		'a' + 1
	on Error do
		print('Correct! An error has occurred 3: {Error:type}\n')
		next
	elseif Z = 9 then
		1 + 'a'
	end
	print('... and on\n')
on Error do
	print('Correct! An error has occurred 4: {Error:type}\n')
end