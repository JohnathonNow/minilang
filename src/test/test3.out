1 + 1 = 2
2 + 1 = 3
hello + 1 = <buffer>
4 + 1 = 5
Correct! An error has occurred 2: MethodError, no matching method found for +(integer, string)
Before X loop
Before Y loop
X = 1, Y = 11
X = 1, Y = 12
X = 1, Y = 13
X = 1, Y = 14
X = 1, Y = 15
After Y loop
Before Y loop
X = 2, Y = 11
X = 2, Y = 12
X = 2, Y = 13
X = 2, Y = 14
X = 2, Y = 15
After Y loop
Before Y loop
X = 3, Y = 11
X = 3, Y = 12
X = 3, Y = 13
Before Y loop
X = 4, Y = 11
X = 4, Y = 12
X = 4, Y = 13
X = 4, Y = 14
X = 4, Y = 15
After Y loop
Before Y loop
X = 5, Y = 11
X = 5, Y = 12
X = 5, Y = 13
X = 5, Y = 14
X = 5, Y = 15
After Y loop
Z = 1
... and on
Z = 2
... and on
Z = 3
... and on
Z = 4
... and on
Z = 5
5!
Correct! An error has occurred 3: ValueError, Offset larger than buffer
Z = 6
... and on
Z = 7
... and on
Z = 8
... and on
Z = 9
Correct! An error has occurred 4: MethodError, no matching method found for +(integer, string)
Z = 10
... and on
